File: IRPlayer-swift/Class/Core/Display/OpenGL/Program/Factory/IRGLProgramDistortionFactory.m
//
//  IRGLProgramDistortionFactory.m
//  IRPlayer
//
//  Created by Phil on 2019/8/23.
//  Copyright © 2019 Phil. All rights reserved.
//

#import "IRGLProgramDistortionFactory.h"
#import <IRPlayer_swift/IRPlayer_swift-Swift.h>

@implementation IRGLProgramDistortionFactory

-(IRGLProgram2D*)createIRGLProgramWithPixelFormat:(IRPixelFormat)pixelFormat withViewprotRange:(CGRect)viewprotRange withParameter:(IRMediaParameter*)parameter{
    return [IRGLProgramFactory createIRGLProgramDistortionWithPixelFormat:pixelFormat viewportRange:viewprotRange parameter:(IRMediaParameter*)parameter];
}

@end

File: Sources/IRPlayer-swift/Class/Core/Display/OpenGL/Program/Factory/IRGLProgramDistortionFactory.swift
//
//  IRGLProgramDistortionFactory.swift
//  IRPlayer-swift
//
//  Created by irons on 2024/9/13.
//

import Foundation

class IRGLProgramDistortionFactory: IRGLProgram2DFactory {

    override func createIRGLProgram(pixelFormat: IRPixelFormat, viewportRange: CGRect, parameter: IRMediaParameter?) -> IRGLProgram2D {
        return IRGLProgramFactory.createIRGLProgramDistortion(pixelFormat: pixelFormat, viewportRange: viewportRange, parameter: parameter)
    }
}
